code_src/staking/dot/fxn_decorator_implementations/accountImplementationUtils.py

```diff
@@ -107,3 +112,38 @@ def getAddressFromMnemonic(self):
                 elif lengthWordInMnemonic or not bip39_validate(self.mnemonic):
                     self.logger.critical("Please check for messing strings.")
                     return False
+
+
+class AccountBalanceForBonding:
+    """
+    Class is to return balance of bonding - verifications
+    TODO: Better explanation
+    # TODO:
+    # * Ideally keep in Utils file instead of here, but need to refactor
+    # to prevent importing accountImplementation.py in Utils
+    """
+
+    def __init__(self, logger, ss58_address):
+        self.logger = logger
+        self.ss58_address = ss58_address
+
+    def __call__(self):
+        return self.getAccountBalanceForBonding()
+
+    def getAccountBalanceForBonding(self):
+        # query balance info for an account
```

Comment:
@luca590 yes you are right i need to sub the existentialDeposit from the totalAccountBalance so that the account will be safe (in fact best practice is 1.5 DOT)

Reply:
@luca590 this is validated using validateAcctBalanceForBonding (fxn_decorator_implementations/substrateCallImplementation.py)